src/libraries/System.Private.CoreLib/src/System/IO/RandomAccess.Unix.cs

```diff
@@ -28,7 +28,9 @@ internal static unsafe int ReadAtOffset(SafeFileHandle handle, Span<byte> buffer
         {
             fixed (byte* bufPtr = &MemoryMarshal.GetReference(buffer))
             {
-                int result = Interop.Sys.PRead(handle, bufPtr, buffer.Length, fileOffset);
+                int result = fileOffset >= 0 ?
```

Comment:
> In my opinion this is breaking the Single-responsibility principle (this method is supposed to read from specific offset from handles that support it).

No more so than the Windows implementation already is, as it already supports this.  It's just that on Windows that's handled by ReadFile, and on Unix we need to emulate it at the call site.  I'm also changing it to do the CanSeek check inside of the method rather than at the call site.